WireDomain/Sources/WireDomain/NotificationService/NotificationSession.swift

```diff
@@ -0,0 +1,97 @@
+//
+// Wire
+// Copyright (C) 2025 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Combine
+import WireAPI
+import WireDataModel
+
+/// Observes pending events, process them and generates new notifications content.
+final class NotificationSession {
+
+    // MARK: - Failure
+
+    enum Failure: Error {
+        case unableToPullPendingEvents(Error)
+    }
+
+    // MARK: - Properties
+
+    private let updateEventsRepository: any UpdateEventsRepositoryProtocol
+    private var subscription: AnyCancellable?
+
+    // MARK: - Object lifecycle
+
+    init(
+        updateEventsRepository: any UpdateEventsRepositoryProtocol,
+        onNotificationContent: @escaping (UNMutableNotificationContent) -> Void
+    ) {
+        self.updateEventsRepository = updateEventsRepository
+        self.subscription = updateEventsRepository.observePendingEvents()
+            .collect() // Collects all the events batches.
+            .map { $0.flatMap { $0 } }
+            .map(generateNotificationContent)
+            .sink(receiveValue: onNotificationContent)
+    }
```

Comment:
What's your reasoning for having a separate publisher to get the events? In my opinion, the benefit of receiving the events directly when calling `pullPendingEvents` is a linear flow of control (call a method, get a result). Here we are mandating thress separate steps in different places (setup observation, call method, teardown observation).